stresstests/test/device_client_helper.js

```diff
@@ -0,0 +1,256 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+const Client = require('azure-iot-device').Client;
+const ModuleClient = require('azure-iot-device').ModuleClient;
+const DeviceSas = require('azure-iot-device').SharedAccessSignature;
+const Registry = require('azure-iothub').Registry;
+const ConnectionString = require('azure-iothub').ConnectionString;
+const anHourFromNow = require('azure-iot-common').anHourFromNow;
+const CommonSas = require('azure-iot-common').SharedAccessSignature;
+const uuid = require('uuid');
+
+const debug = require('debug')('stresstests:DeviceClientHelper');
+
+/**
+ * Helps manage the lifecycle of a device client (Client) or module client (ModuleClient).
+ */
+class DeviceClientHelper {
+  /**
+   * Construct a DeviceClientHelper instance.
+   * @param {string} [iotHubConnectionString] - The connection string to the IoT Hub.
+   *   If not provided, the environment variable IOTHUB_CONNECTION_STRING will be used.
+   * @public
+   */
+  constructor(iotHubConnectionString) {
+    [this.client, this.deviceId, this.moduleId] = [null, null, null];
+    this._iotHubConnectionString = iotHubConnectionString ?
+      iotHubConnectionString :
+      process.env.IOTHUB_CONNECTION_STRING;
+    this._hubHostName =
+      ConnectionString.parse(this._iotHubConnectionString).HostName;
+    this._registry = Registry.fromConnectionString(this._iotHubConnectionString);
+  }
+
+  /**
+   * Closes the client, disposes of the client, and removes the device from the IoT Hub.
+   * @public
+   */
+  async disposeClient() {
+    if (!this.client) {
+      debug('attempted to dispose a non-exiting client')
+      return;
+    }
+    debug('disposing client');
+    const [client, deviceId] = [this.client, this.deviceId];
+    [this.client, this.deviceId, this.moduleId] = [null, null, null];
+    await Promise.all([
+      client.close().then(() => debug('client closed')),
+      this._registry.delete(deviceId).then(() => debug('device id removed from registry'))
+    ]);
+  }
+
+  /**
+   * Registers a device with the IoT Hub and then creates a corresponding device
+   * client (Client) that authenticates using a shared access signature.
+   * @param {function} transportCtor - The constructor of the device transport to use.
+   * @param {string} [modelId] - An optional model ID to use for the device.
```

Comment:
>optional model ID to [](http://example.com/codeflow?start=36&length=20)

PNP model id, right?  Maybe use a more specific name for the parameter

Reply:
Actually, I'll just remove this because we never shipped the native PnP implementation